gap/congruences/congsemigraph.gi

```diff
@@ -0,0 +1,271 @@
+############################################################################
+##
+##  congsemigraph.gi
+##  Copyright (C) 2021                       Marina Anagnostopoulou-Merkouri
+##                                                            James Mitchell
+##
+##  Licensing information can be found in the README file of this package.
+##
+############################################################################
+
+BindGlobal("SEMIGROUPS_IsHereditarySubset",
+function(S, H)
+  local out, h, v, D;
+  D := GraphOfGraphInverseSemigroup(S);
+  if IsMultiDigraph(D) then
+    ErrorNoReturn(D, " must not have multiple edges");
+  fi;
+  out := OutNeighbours(D);
+  for h in H do
+    for v in out[h] do
+      if not (v in H) then
+        return false;
+      fi;
+    od;
+  od;
+  return true;
+end);
+
+BindGlobal("SEMIGROUPS_IsValidWSet",
+function(S, H, W)
+  local w, c, v, out, D;
+  if not SEMIGROUPS_IsHereditarySubset(S, H) then
```

Comment:
Are there any further prerequisites for `H` and `W`, like shouldn't they be duplicate-free, and/or maybe sorted?